src/transformers/models/opt/modeling_opt.py

```diff
@@ -191,71 +193,88 @@ def forward(
             # if encoder bi-directional self-attention `past_key_value` is always `None`
             past_key_value = (key_states, value_states)
 
-        proj_shape = (bsz * self.num_heads, -1, self.head_dim)
-        query_states = self._shape(query_states, tgt_len, bsz).view(*proj_shape)
-        key_states = key_states.view(*proj_shape)
-        value_states = value_states.view(*proj_shape)
+        if fast_attention:
+            key_states_fast = key_states.view(bsz, -1, self.embed_dim)
+            value_states_fast = value_states.view(bsz, -1, self.embed_dim)
+            seqlen = key_states_fast.shape[1]
+            query_states_fast = torch.nested_tensor(torch.unbind(query_states, dim=0))
+            key_states_fast = torch.nested_tensor(torch.unbind(key_states_fast, dim=0))
+            value_states_fast = torch.nested_tensor(torch.unbind(value_states_fast, dim=0))
```

Comment:
won't this result in padding within the nested tensors?

i.e. if query_states is a padded rectangular tensor, calling unbind on it will produce sequences padded to the same length, so we won't be taking advantage of nested tensors to reduce padding. Or am I missing something?

Reply:
I'm assuming only 0 padding tensor inputs for now. This is a hack just to make those tensors into NestedTensors because currently FlashAttention SDP requires NestedTensor. If FlashAttn SDP supported regular tensor I would just remove these entirely.

Reply:
gotcha, thanks for the clarification :)